src/components/with_search.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import Search from "./search/search";
+import Column from "./column/column";
+const withSearch = Component => ({
+  searchRenderer = Search,
+  showSearch,
+  filterItems,
+  searchIcon,
+  searchValue,
+  messages,
+  ...others
+}) => {
+  const SearchRenderer = searchRenderer;
+  return showSearch ? (
+    <Column>
```

Comment:
I'm not sure I fully agree that withSearch should handle the layout of the column - your call.

Reply:
yes, you are right - i removed column from HOC and changed approach to use children, please take a look.